scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TensorFlowDoFns.scala

```diff
@@ -112,3 +111,100 @@ private[tensorflow] class SavedBundlePredictDoFn[T, V](
   }
 
 }
+
+/**
+ * Uses sigDef from the graph to translate input and output tensors according to exported signature.
+ * For example:
+ *
+ * input:
+ *  age -> input_tensor:0
+ *  lang -> input_tensor:1
+ *
+ * output:
+ *  probabilities -> softmax:0
+ *
+ * Caller:
+ *  input: (age -> 50, lang -> SE)
+ *  fetchOps: probabilities
+ *
+ * At prediction time, age and lang will be translated to their graph tensor names.
+ * And the fetch op for probabilities will also be mapped accordingly.
+ *
+ * Note: right now this only loads the default sigDef from tensorflow.
+ */
+private[tensorflow] class SavedBundlePredictWithSignatureTranslationDoFn[T, V](
+  uri: String,
+  options: TensorFlowModel.Options,
+  fetchOp: Seq[String],
+  inFn: T => Map[String, Tensor[_]],
+  outFn: (T, Map[String, Tensor[_]]) => V
+) extends PredictDoFn[T, V, TensorFlowModel](fetchOp, inFn, outFn) {
+  @transient private lazy val log = LoggerFactory.getLogger(this.getClass)
+
+  private lazy val loadModel = new Function[String, TensorFlowModel] {
```

Comment:
`loadModel` is used a lot maybe smth like:

```scala
  @transient private lazy val model = getResource
    .computeIfAbsent(
      uri,
      new Function[String, TensorFlowModel] {
        override def apply(t: String): TensorFlowModel =
          Models
            .tensorFlow(uri, options)
            .get(Duration.ofDays(Integer.MAX_VALUE))
      }
    )
```

I added `@transient` here as well because i think we had serialization issues before.